### ISSUE TITLE
[client] client undefined using withMutations

### ISSUE THREAD

**Issue by CPatchane** [cozy/cozy-client#198]
Here is my codebase:

```jsx
const TodoAdd = props => (
  <button onClick={props.createDocument}>test</button>
)
export default withMutations()(TodoAdd)
```

And I got this error:
```
TypeError: "client is undefined, can't access property "create" of it"
	Wrapper http://localhost:8888/app8aa32fd4e97c1db49c24.bundle.js:53341:11
	ProxyComponent http://localhost:8888/app8aa32fd4e97c1db49c24.bundle.js line 101695 > eval:5:7
	constructClassInstance http://localhost:8888/app8aa32fd4e97c1db49c24.bundle.js:94931:18
	updateClassComponent http://localhost:8888/app8aa32fd4e97c1db49c24.bundle.js:96653:7
	beginWork http://localhost:8888/app8aa32fd4e97c1db49c24.bundle.js:97252:16
	performUnitOfWork http://localhost:8888/app8aa32fd4e97c1db49c24.bundle.js:99578:12
	workLoop http://localhost:8888/app8aa32fd4e97c1db49c24.bundle.js:99616:24
	renderRoot http://localhost:8888/app8aa32fd4e97c1db49c24.bundle.js:99695:7
	performWorkOnRoot http://localhost:8888/app8aa32fd4e97c1db49c24.bundle.js:100549:7
	performWork http://localhost:8888/app8aa32fd4e97c1db49c24.bundle.js:100457:7
	performSyncWork http://localhost:8888/app8aa32fd4e97c1db49c24.bundle.js:100429:3
	requestWork http://localhost:8888/app8aa32fd4e97c1db49c24.bundle.js:100317:5
	scheduleWork http://localhost:8888/app8aa32fd4e97c1db49c24.bundle.js:100111:5
	enqueueSetState http://localhost:8888/app8aa32fd4e97c1db49c24.bundle.js:94785:5
	setState http://localhost:8888/app8aa32fd4e97c1db49c24.bundle.js:106519:3
	onQueryChange http://localhost:8888/app8aa32fd4e97c1db49c24.bundle.js:50486:7
	notifyObservers http://localhost:8888/app8aa32fd4e97c1db49c24.bundle.js:50271:36
	notifyObservers http://localhost:8888/app8aa32fd4e97c1db49c24.bundle.js:50270:43
	handleStoreChange http://localhost:8888/app8aa32fd4e97c1db49c24.bundle.js:50225:9
	dispatch http://localhost:8888/app8aa32fd4e97c1db49c24.bundle.js:108328:7
	dispatch http://localhost:8888/app8aa32fd4e97c1db49c24.bundle.js:50066:14
	_callee6$ http://localhost:8888/app8aa32fd4e97c1db49c24.bundle.js:49532:17
	tryCatch http://localhost:8888/app8aa32fd4e97c1db49c24.bundle.js:108585:37
	invoke http://localhost:8888/app8aa32fd4e97c1db49c24.bundle.js:108823:22
	method http://localhost:8888/app8aa32fd4e97c1db49c24.bundle.js:108637:16
	step http://localhost:8888/app8aa32fd4e97c1db49c24.bundle.js:1161:30
	step http://localhost:8888/app8aa32fd4e97c1db49c24.bundle.js:1172:13
	run http://localhost:8888/app8aa32fd4e97c1db49c24.bundle.js:4182:22
	notify http://localhost:8888/app8aa32fd4e97c1db49c24.bundle.js:4199:30
	flush http://localhost:8888/app8aa32fd4e97c1db49c24.bundle.js:2940:9
```

**Comment by y-lohse** [comment]
The error is pretty bad and we should improve it.

However, mutations are mostly helpful when dealing with `Query` components — I don't think they're even supposed to work without one.

If you just want to update a document, you can interact with your doctype through the [corresponding collection](https://github.com/cozy/cozy-client/blob/schema-fixes/packages/cozy-stack-client/src/DocumentCollection.js#L156). Maybe something like `onClick={() => client.collection('io.cozy.todos').create(data)}` should work.

**Comment by CPatchane** [comment]
Fixed by #200

**Comment by y-lohse** [comment]
Right, I was missing that context from the initial issue description :)
